ui/src/main/webapp/src/app/registered-application/register-application-form.component.ts

```diff
@@ -63,6 +65,32 @@ export class RegisterApplicationFormComponent extends FormComponent implements O
             this._notificationService.error(utils.getErrorMessage(response));
         };
         this.multipartUploader.onAfterAddingFile = () => this.registerUploaded();
+        this.multipartUploader.onWhenAddingFileFailed = (item: FileLikeObject, filter: FilterFunction, options) => {
+            let msg;
+            if (filter["rejectMessage"])
+                msg = formatString(filter["rejectMessage"], item.name);
+            else
+                switch (filter.name) {
+                    case "onlyJavaArchives": msg = `File was rejected: '${item.name}'\nOnly Java archives are accepted (${filter["suffixes"].join(", ")})`; break;
+                    case 'queueLimit': msg = "Maximum number of queued files reached."; break;
+                    case 'fileSize': msg = `File '${item.name}' is too large.`; break;
+                    case 'fileType': msg = `File '${item.name}' is of invalid file type.`; break;
+                    case 'mimeType': msg = `File '${item.name}' is of invalid MIME type.`; break;
+                    default: `File rejected for uploading: '${item.name}'`; break;
+                }
+            this._notificationService.error(msg);
+        }
+
+        let suffixes = ".war .ear .jar .sar .har .rar".split(" ");
```

Comment:
I think all of those are actually just zip (even rar, because it is really zip file called a "resource archive") files and I believe Windup will handle them.

It doesn't handle gzip, though.